students/M_Sunday/lesson08/circle.py

```diff
@@ -0,0 +1,78 @@
+#!/usr/bin/python
+from math import pi
+
+
+class Circle(object):
+
+    @staticmethod
+    def entry_check(the__radius):
+        if isinstance(the__radius, (str, list, tuple, dict)):
+            raise TypeError("Radius/Diameter entry must be a single, positive, and "
+                            "non-string value")
+        else:
+            if the__radius < 0:
+                raise ValueError("Radius/Diameter must be greater than 0")
+            else:
+                return the__radius
+
+    def __init__(self, the_radius):
+        self._radius = self.entry_check(the_radius)
+
+    def __repr__(self):
+        return "Circle({})".format(self._radius)
+
+    def __str__(self):
+        return "Circle with radius: {0:.6f}".format(float(self._radius))
+
+    def __add__(self, other):
+        new_circle = self._radius + other._radius
+        return Circle(new_circle)
+
+    def __rmul__(self, other):
+        new_circle = self._radius * other
+        return Circle(new_circle)
+
+    def __mul__(self, other):
+        new_circle = self._radius * other
+        return Circle(new_circle)
+
+    def __lt__(self, other):
+        return self._radius < other._radius
+
+    def __gt__(self, other):
+        return self._radius > other._radius
+
+    def __eq__(self, other):
+        return self._radius == other._radius
```

Comment:
You can use the functools.total_ordering decorator from the Python standard library and only define __eq__ and __lt__, otherwise should define the other comparion operators as well